### PR TITLE
Rescheduling fixes

### PR DESCRIPTION
**Pull request by abnegate**
## What does this PR do?

Scheduling fixes:
- Throw on update is no schedule exists
- Throw on update if no recipients and status is not draft
- Disable schedules when updating message to draft or processing
- Fix `scheduledAt` not updated on update routes

## Test Plan

(Write your test plan here. If you changed any code, please provide us with clear instructions on how you verified your changes work. Screenshots may also be helpful.)

## Related PRs and Issues

- (Related PR or issue)

## Checklist

- [ ] Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?
- [ ] If the PR includes a change to an API's metadata (desc, label, params, etc.), does it also include updated API specs and example docs?

### CODE REVIEW

File: app/controllers/api/messaging.php [hunk 1]
```diff
@@ -3251,19 +3251,64 @@
...
+                throw new Exception(Exception::SCHEDULE_NOT_FOUND);
+            }
+
+            if (!\is_null($scheduledAt)) {
+                $schedule
+                    ->setAttribute('resourceUpdatedAt', DateTime::now())
+                    ->setAttribute('schedule', $scheduledAt)
+                    ->setAttribute('active', $status === MessageStatus::SCHEDULED);
+            } else {
+                $schedule->setAttribute('active', $status === MessageStatus::SCHEDULED);
```
Comment (eldadfux) [inline app/controllers/api/messaging.php:3306]: Should this be the same as the above?

File: app/controllers/api/messaging.php [hunk 2]
```diff
@@ -3613,38 +3679,6 @@
...
-                    ->setAttribute('resourceUpdatedAt', DateTime::now())
-                    ->setAttribute('schedule', $scheduledAt)
-                    ->setAttribute('active', $status === MessageStatus::SCHEDULED);
-
-                $dbForConsole->updateDocument('schedules', $schedule->getId(), $schedule);
-            }
-
-            $message->setAttribute('scheduleId', $schedule->getId());
-        }
-
```
Comment (eldadfux) [inline app/controllers/api/messaging.php:3647]: Can we add a test for canceling the schedule?

File: app/controllers/api/messaging.php (hunk 1 above)
Comment (abnegate) [reply app/controllers/api/messaging.php:3306]: No, if you didn't changed the scheduled at time, it should just be activated/deactivated

File: app/controllers/api/messaging.php (hunk 2 above)
Comment (abnegate) [reply app/controllers/api/messaging.php:3647]: Will do :+1:

File: app/controllers/api/messaging.php [hunk 3]
```diff
@@ -3251,19 +3251,68 @@
...
                 throw new Exception(Exception::MESSAGE_ALREADY_FAILED);
+            case MessageStatus::SCHEDULED:
+                if (\is_null($scheduledAt) && \is_null($currentScheduledAt)) {
+                    throw new Exception(Exception::MESSAGE_MISSING_SCHEDULE);
+                }
+                break;
         }
 
-        if (!\is_null($message->getAttribute('scheduledAt')) && $message->getAttribute('scheduledAt') < new \DateTime()) {
+        if (!\is_null($currentScheduledAt) && $currentScheduledAt < new \DateTime()) {
```
Comment (stnguyen90) [inline app/controllers/api/messaging.php:3274]: I don't think comparing a string with a DateTime() works. 

image

File: app/controllers/api/messaging.php [hunk 4]
```diff
@@ -3251,19 +3251,68 @@
...
                 throw new Exception(Exception::MESSAGE_ALREADY_PROCESSING);
             case MessageStatus::SENT:
                 throw new Exception(Exception::MESSAGE_ALREADY_SENT);
             case MessageStatus::FAILED:
                 throw new Exception(Exception::MESSAGE_ALREADY_FAILED);
+            case MessageStatus::SCHEDULED:
+                if (\is_null($scheduledAt) && \is_null($currentScheduledAt)) {
+                    throw new Exception(Exception::MESSAGE_MISSING_SCHEDULE);
+                }
+                break;
```
Comment (stnguyen90) [inline app/controllers/api/messaging.php:3271]: What if the message was draft and then I update it to be scheduled, but I don't pass `$scheduledAt`?

**Review by stnguyen90 — CHANGES_REQUESTED** [review]

File: app/controllers/api/messaging.php (hunk 4 above)
Comment (eldadfux) [reply app/controllers/api/messaging.php:3271]: Should schedule even be a status??? I really doubt this.

Comment (abnegate) [reply app/controllers/api/messaging.php]: Resolved by https://github.com/appwrite/appwrite/pull/7670

**Review by eldadfux — APPROVED** [review]

**Comment by stnguyen90** [comment]
> It was bad DX to allow explicitly passing the status, and we only needed draft as a bool and the existing sheduledAt params to determine the status, so we decided to remove it.

**Pull request merged**